src/data/finset/lattice.lean

```diff
@@ -961,6 +961,48 @@ lemma induction_on_min [decidable_eq α] {p : finset α → Prop} (s : finset α
 
 end max_min
 
+section max_min_induction_value
+
+variables {ι : Type*} [linear_order α] [linear_order β]
+
+/-- Induction principle for `finset`s in any type from which a given function `f` maps to a linearly
+ordered type : a predicate is true on all `s : finset α` provided that:
+
+* it is true on the empty `finset`,
+* for every `s : finset α` and an element `a` such that for elements of `s` denoted by `x` we have
+  `f x ≤ f a`, `p s` implies `p (insert a s)`. -/
+@[elab_as_eliminator]
+lemma induction_on_max_value [decidable_eq ι] (f : ι → α)
+  {p : finset ι → Prop} (s : finset ι) (h0 : p ∅)
+  (step : ∀ a s, a ∉ s → (∀ x ∈ s, f x ≤ f a) → p s → p (insert a s)) : p s :=
+begin
+  induction s using finset.strong_induction_on with s ihs,
+  rcases (s.image f).eq_empty_or_nonempty with hne|hne,
+  { simp only [image_eq_empty] at hne,
+    simp only [hne, h0] },
+  { have H : (s.image f).max' hne ∈ (s.image f), from max'_mem (s.image f) hne,
+    simp only [mem_image, exists_prop] at H,
+    rcases H with ⟨a, has, hfa⟩,
+    rw ← insert_erase has,
+    refine step _ _ (not_mem_erase a s) (λ x hx, _) (ihs _ $ erase_ssubset has),
+    rw hfa,
+    exact le_max' _ _ (mem_image_of_mem _ $  mem_of_mem_erase hx) }
+end
```

Comment:
Is this true on multisets too? If so, you can probably prove the finset version from the multiset version in a few lines.

Reply:
Ah, there's no `multiset.max` so that makes this awkward. Fine as is then.

Reply:
It is true for multisets too and I'd be happy to make this deduction. Currently, I can't seem to find a nice way to take the maximum of a multiset, any suggestions? @eric-wieser